pkg/client/unversioned/request.go

```diff
@@ -149,6 +150,10 @@ func (r *Request) Resource(resource string) *Request {
 		r.err = fmt.Errorf("resource already set to %q, cannot change to %q", r.resource, resource)
 		return r
 	}
+	if ok, msg := validation.ValidatePathSegmentName(resource, false); !ok {
```

Comment:
agree on guarding the data HPA uses, but Request's job is to take logical data and turn it into a URL... it should guard against data that breaks its ability to do that